src/hotspot/share/runtime/handshake.cpp

```diff
@@ -305,14 +325,14 @@ void HandshakeOperation::do_handshake(JavaThread* thread) {
 }
 
 void Handshake::execute(HandshakeClosure* hs_cl) {
-  HandshakeOperation cto(hs_cl, NULL);
+  HandshakeOperation cto(hs_cl, NULL, Thread::current());
```

Comment:
It is unfortunate that we need to manifest the current thread here (and elsewhere).

Reply:
Yes, I did that since I did not want to change the API.
I was thinking of:
Handshake::execute(HandshakeClosure* hs_cl, Thread* requester = NULL) 
And then go over calls site and see if we have the current thread in a variable.
But I felt that would be better off in another change, if wanted.